radixdlt/src/main/java/com/radixdlt/consensus/EventCoordinator.java

```diff
@@ -67,57 +77,80 @@ public EventCoordinator(
 		this.pacemaker = Objects.requireNonNull(pacemaker);
 		this.vertexStore = Objects.requireNonNull(vertexStore);
 		this.proposerElection = Objects.requireNonNull(proposerElection);
-		this.self = Objects.requireNonNull(self);
+		this.quorumRequirements = Objects.requireNonNull(quorumRequirements);
+		this.selfAddress = Objects.requireNonNull(selfAddress);
+		this.selfKey = Objects.requireNonNull(selfKey);
+		if (!selfAddress.getKey().equals(selfKey.getPublicKey())) {
+			throw new IllegalArgumentException("Address and key mismatch: " + selfAddress + " != " + selfKey);
+		}
 	}
 
-	private void processNewRound(Round round) {
-		log.debug("Processing new round: " +  round);
+	private void processNewView(View view) {
+		log.debug("Processing new view: " + view);
 		// only do something if we're actually the leader
-		if (!proposerElection.isValidProposer(self, round)) {
+		if (!proposerElection.isValidProposer(selfAddress.getUID(), view)) {
 			return;
 		}
 
-		Vertex proposal = proposalGenerator.generateProposal(this.pacemaker.getCurrentRound());
+		Vertex proposal = proposalGenerator.generateProposal(this.pacemaker.getCurrentView());
 
 		// TODO: Handle empty proposals
 		if (proposal.getAtom() != null) {
 			this.networkSender.broadcastProposal(proposal);
+			log.info("Starting view " + view + " with proposal " + proposal);
 		}
 	}
 
 	public void processVote(Vote vote) {
-		// only do something if we're actually the leader for the next round
-		if (!proposerElection.isValidProposer(self, vote.getVertexMetadata().getRound().next())) {
-			log.warn(String.format("Ignoring confused vote %s for %s", vote.hashCode(), vote.getVertexMetadata().getRound()));
+		// only do something if we're actually the leader for the next view
+		if (!proposerElection.isValidProposer(selfAddress.getUID(), vote.getVertexMetadata().getView().next())) {
+			log.warn(String.format("Ignoring confused vote %s for %s", vote.hashCode(), vote.getVertexMetadata().getView()));
 			return;
 		}
 
-		// accumulate votes into QCs
-		// TODO assumes a single node network for now
-		QuorumCertificate qc = new QuorumCertificate(vote, vote.getVertexMetadata());
-		this.safetyRules.process(qc);
-		this.vertexStore.syncToQC(qc);
-		this.pacemaker.processQC(qc.getRound())
-			.ifPresent(this::processNewRound);
+		// accumulate votes into QCs in store
+		Optional<QuorumCertificate> potentialQc = this.vertexStore.insertVote(vote, this.quorumRequirements);
```

Comment:
Don't think vote accumulation should be occurring in the vertexStore

Reply:
Agreed, where do you propose putting it? Should probably just go in its own `PendingVotes` class.

Reply:
Yes, that sounds about right.